plugins/net_plugin/net_plugin.cpp

```diff
@@ -1492,10 +1735,52 @@ namespace eosio {
                        ("lib", last_handshake.last_irreversible_block_num)
                        ("head", last_handshake.head_num)("id", last_handshake.head_id.str().substr(8,16)) );
             c->enqueue( last_handshake );
+            // only send request to peers allowed connection
+            // every 100 handshakes will send a request
+            if(c->is_peers_connection() && c->sent_handshake_count % my_impl->address_request_frequency == 1) {
+                //first time send push request
+                if(c->sent_handshake_count == 1) {
+                    c->enqueue_address_request(push);
+                } else {
+                    c->enqueue_address_request(pull);
+                }
+            }
          }
       });
    }
 
+   void connection::send_address_request(request_type_enum request_type) {
+       strand.post( [c = shared_from_this(), request_type]() {
+            c->enqueue_address_request(request_type);
+       });
+   }
+
+   // called from connection strand
+   void connection::enqueue_address_request(request_type_enum request_type) {
+       // only send request to peers that support the protocol
+       if(protocol_version >= proto_address_sync) {
+           peer_dlog(this, "Sending address request type ${type}", ("type", request_type_str(request_type)));
+           address_request_message request_msg;
+
+           request_msg.request_type = request_type;
+           // request push type will send addresses to other node
+           if (request_type == push) {
+               std::unordered_set<std::string> addresses = my_impl->address_master->get_addresses();
```

Comment:
if `addresses.size()` <= `my_impl->max_addresses_per_request` you can just `std::move(addresses)` into `request_msg.addresses`. Otherwise you can erase a distance to reduce the size of addresses and then move. Or better yet pass `my_impl->max_addresses_per_request` to `get_addresses()` and only return a max of that many.